### PR TITLE
Add RPC function to regenerate mint pool if incorrect indexes used

### PR DESCRIPTION
**Pull request by riordant**
## PR intention
- FIxes an issue where wallets, generated with HD mint prior to 0.13.8.2, have mints that are always stuck in "pending" on the Sigma page. This is due to incorrect indexes being generated for the mint pool database functions in the 0.13.8.1 release. The fix is simply to regenerate the mint pool from the known values, and add with the correct indexes.

- Also adds increased logging level generally for HD mint code.

## Code changes brief
- Add RPC function `regeneratemintpool`
- Modification of `RegenerateMintPoolEntry` to return a uint256 pair - `<hashPubcoin, hashSerial>`

### CODE REVIEW

File: src/hdmint/wallet.cpp [hunk 1]
```diff
@@ -142,12 +151,30 @@ bool CHDMintWallet::LoadMintPoolFromDB()
...
         mintPool.Add(mintPoolPair);
+    }
 
     return true;
 }
 
+bool CHDMintWallet::GetSerialForPubcoin(const std::vector<std::pair<uint256, GroupElement>>& serialPubcoinPairs, const uint256& hashPubcoin, uint256& hashSerial)
+{
+    bool fFound = false;
+    for(auto serialPubcoinPair : serialPubcoinPairs){
```
Comment (a-bezrukov) [inline src/hdmint/wallet.cpp]: May it be a const ref, not a copy?

File: src/hdmint/wallet.cpp [hunk 2]
```diff
@@ -142,12 +151,30 @@ bool CHDMintWallet::LoadMintPoolFromDB()
...
+        GroupElement pubcoin = serialPubcoinPair.second;
```
Comment (a-bezrukov) [inline src/hdmint/wallet.cpp]: Same here

Comment (riordant) [reply src/hdmint/wallet.cpp]: Agreed.

File: src/hdmint/wallet.cpp (hunk 1 above)
Comment (riordant) [reply src/hdmint/wallet.cpp]: Agreed

File: src/hdmint/wallet.cpp (hunk 2 above)
Comment (riordant) [reply src/hdmint/wallet.cpp]: Could you clarify what you mean here?

Comment (a-bezrukov) [reply src/hdmint/wallet.cpp]: pubcoin is a copy now. I think making it ``` GroupElement const & pubcoin = serialPubcoinPair.second``` will be better performance-wise.

Comment (levonpetrosyan93) [reply src/hdmint/wallet.cpp]: I think it can be like this  const GroupElement& pubcoin.

Comment (levonpetrosyan93) [reply src/hdmint/wallet.cpp]: Will be more readable if we keep this line, only make it const reference.

**Pull request merged**